packages/block-library/src/categories/edit.js

```diff
@@ -116,7 +116,13 @@ class CategoriesEdit extends Component {
 					{ __( 'Categories' ) }
 				</label>
 				<select id={ selectId } className="wp-block-categories__dropdown">
-					{ categories.map( ( category ) => this.renderCategoryDropdownItem( category, 0 ) ) }
+					{ categories.filter(
```

Comment:
This made me think we could filter inside `getCategories` method to make it easier to follow here.

I will test this PR by tomorrow.

Reply:
I made a new check in the logic of the code and it looks like the filtering was already being accounted in the logic. The existing logic only had a bug when accessing a prop that caused the filtering to not be performed. The fix was simplified.

Reply:
This is nice, thanks for investigating further, testing your changes soon.